consensus/event_loop_test.go

```diff
@@ -418,6 +418,50 @@ func TestProcessValidatorTimeout(t *testing.T) {
 	fc.Stop()
 }
 
+func TestTimeoutScheduledTaskExecuted(t *testing.T) {
+	oneHourInNanoSecs := time.Duration(3600000000000)
+	oneMilliSecondInNanoSeconds := time.Duration(1000000)
+
+	dummyBlock := createDummyBlock()
+
+	fc := NewFastHotStuffEventLoop()
+	err := fc.Init(oneHourInNanoSecs, oneMilliSecondInNanoSeconds, dummyBlock, createDummyValidatorSet())
+	require.NoError(t, err)
+
+	// Start the event loop
+	fc.Start()
+
+	// Wait for the timeout signal to be sent
+	timeoutSignal := <-fc.ConsensusEvents
+
+	// Confirm that the timeout signal is for the the expected view
+	require.Equal(t, timeoutSignal.EventType, ConsensusEventTypeTimeout)
+	require.Equal(t, timeoutSignal.View, dummyBlock.GetView()+1)
+	require.Equal(t, timeoutSignal.BlockHash.GetValue(), dummyBlock.GetBlockHash().GetValue())
+
+	// Confirm that the timeout and block construction scheduled tasks are no longer running
+	require.False(t, fc.nextTimeoutTask.IsScheduled())
+	require.False(t, fc.nextBlockConstructionTask.IsScheduled())
+
+	// Advance the view, which should reset the timeout scheduled task
+	fc.AdvanceView()
+
```

Comment:
Can we add some testing in here to verify that the the tasks are rescheduled?

Reply:
When line 450 returns, it guarantees that the tasks were rescheduled and have executed. 

I think it would be difficult to check externally here what params the tasks were rescheduled with without adding more complexity to the task implementations. Not sure the extra test coverage is worth it here